plugin/pkg/admission/storage/persistentvolume/label/admission.go

```diff
@@ -140,7 +147,17 @@ func (l *persistentVolumeLabel) Admit(ctx context.Context, a admission.Attribute
 			} else {
 				values = []string{v}
 			}
-			requirements = append(requirements, api.NodeSelectorRequirement{Key: k, Operator: api.NodeSelectorOpIn, Values: values})
+
+			// separate topology requirements based on deprecated vs stable zone/region labels
+			// all other labels apply to both requirements
+			if k == v1.DeprecatedLabelZoneFailureDomain || k == v1.DeprecatedLabelZoneRegion {
+				deprecatedTopologyReqs = append(deprecatedTopologyReqs, api.NodeSelectorRequirement{Key: k, Operator: api.NodeSelectorOpIn, Values: values})
```

Comment:
nit: create a helper function for generating `NodeSelectorRequirement` instead of doing these struct initializations everywhere. Improves maintainability

Reply:
done, but didn't re-use for the PV label admission controller cause the logic varies there vs dynamically provisioned volumes.